packages/studio-ui/src/components/OpenLivePreviewButton.tsx

```diff
@@ -1,19 +1,81 @@
+import { PagesJsState } from "@yext/studio-plugin";
+import useStudioStore from "../store/useStudioStore";
+import classNames from "classnames";
+import { Result } from "true-myth";
+import { useCallback } from "react";
+
+// The URL of the Landing Page of the PagesJS Dev Server. Port 5173 is hardcoded for now as it will be the PagesJS Dev port in most cases.
+export const PAGES_JS_LANDING_PAGE = "http://localhost:5173";
+
 /**
- * OpenLivePreviewButton is a button that when clicked, opens the
- * pages development server index page in a new tab.
- * Port 5173 hardcoded for now as it will be the pages dev port in most cases
+ * The button rendered by this Component opens a PagesJS-powered Preview of the current Page in a new tab.
+ * If the Page is an Entity Template, the Preview will use the currently selected Entity Data.
  */
-export default function OpenLivePreviewButton(): JSX.Element | null {
+export default function OpenLivePreviewButton(): JSX.Element {
+  let previewDisabled;
+  let livePreviewUrl;
+
+  const [activePageName, activePageState, activeEntityData] = useStudioStore(
+    (store) => [
+      store.pages.activePageName as string,
+      store.pages.getActivePageState(),
+      store.pages.getActiveEntityData(),
+    ]
+  );
+
+  const isActivePagesJSPage = !!activePageState?.pagesJS;
+  if (isActivePagesJSPage) {
+    const pagesJSState = activePageState.pagesJS as PagesJsState;
+    const livePreviewUrlResult = getLivePreviewUrl(
+      pagesJSState,
+      activePageName,
+      activeEntityData
+    );
+    previewDisabled = livePreviewUrlResult.isErr;
+    livePreviewUrl = livePreviewUrlResult.isOk && livePreviewUrlResult.value;
+  } else {
+    previewDisabled = true;
+  }
+
+  const onClick = useCallback(() => {
+    livePreviewUrl && window.open(livePreviewUrl, "_blank");
+  }, [livePreviewUrl]);
+  const buttonClasses = classNames(
+    "rounded-md px-2 py-1 flex items-center gap-x-2 text-white",
+    {
+      "bg-gray-400": previewDisabled,
+      "bg-blue-600 shadow-md hover:bg-blue-700 hover:shadow-lg":
+        !previewDisabled,
+    }
+  );
+
   return (
     <div className="relative inline-block">
-      <a
-        className="rounded-md text-white bg-blue-600 px-2 py-1 flex items-center gap-x-2 shadow-md hover:bg-blue-700 hover:shadow-lg"
-        href="http://localhost:5173/"
-        target="_blank"
-        rel="noreferrer"
+      <button
```

Comment:
Just a thought-- but if a user has a non-pagesJS repo, should we just not display the button at all since it will never become enabled for that repo?